### PR TITLE
Add composite answer type

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
This PR add the composite answer type which allows for a parent answer to have multiple child answers.

### How to review 
Tests pass any the new type and input make sense.

### CODE REVIEW

**Comment by SamGodwin2** [comment]
[Piping Date Ranges (5)](https://trello.com/c/pjFZp9OC/315-piping-date-ranges-5)

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eq-author-graphql-schema",
-  "version": "0.17.0",
```
Comment (samiwel) [inline package.json]: Looks like this is a few versions behind now. Worth publishing a new minor version (e.g. 0.18.0 once composite answers are introduced?

File: index.js [hunk 2]
```diff
@@ -95,6 +95,19 @@ type MultipleChoiceAnswer implements Answer {
...
+  description: String
+  guidance: String
+  qCode: String
+  label: String
+  type: AnswerType!
+  mandatory: Boolean @deprecated(reason: "Use \`properties\` instead.")
+  page: QuestionPage
+  childAnswers: [BasicAnswer]!
+  properties: JSON
+}
```
Comment (samiwel) [inline index.js]: We may have already discussed this before, but probably worth capturing here anyway... was any consideration given to making the type a bit more explicit for date ranges?

e.g.

```graphql
type DateRangeAnswer implements Answer {
  id: ID!
  ...
  to: BasicAnswer!
  from: BasicAnswer!
}
```

`CompositeAnswer` gives us more flexibility, but the `childAnswers` field only ensures that the array is not null, whereas the example I provide above would ensure that both `to` and `from` are both not null.

Comment (SamGodwin2) [reply index.js]: Yeah iirc when we were discussing it we ended up on the side of more flexibility as I think there are eventually requirements for more composite answer types? In which case this composite type can be reused. I'm not sure how crucial it is that the schema validates to and from are !null as the api will be enforcing it in the resolvers?

**Pull request merged**